### ISSUE TITLE
Issue in Flutter SDK 3.10

### ISSUE THREAD

**Issue by sarmiladhandapani** [rwbr/flutter_neat_and_clean_calendar#55]
../../.pub-cache/hosted/pub.dev/macos_ui-1.12.2/lib/src/fields/text_field.dart:108:35: Error: The parameter 'details' of the method '_TextFieldSelectionGestureDetectorBuilder.onSingleTapUp' has type 'TapUpDetails', which does not match the corresponding type, 'TapDragUpDetails', in the overridden method, 'TextSelectionGestureDetectorBuilder.onSingleTapUp'.
 - 'TapUpDetails' is from 'package:flutter/src/gestures/tap.dart' ('../../Development/flutter/packages/flutter/lib/src/gestures/tap.dart').
 - 'TapDragUpDetails' is from 'package:flutter/src/widgets/tap_and_drag_gestures.dart' ('../../Development/flutter/packages/flutter/lib/src/widgets/tap_and_drag_gestures.dart').
Change to a supertype of 'TapDragUpDetails', or, for a covariant parameter, a subtype.
  void onSingleTapUp(TapUpDetails details) {

**Comment by rwbr** [comment]
Well, could you please add more information so that I understand in which way your problem is related to this project?

**Comment by LZaruba** [comment]
Hey,

I'm facing the same error. Minimal reproducible code is to take the autogenerated default flutter app and add the calendar as shown in the example. Once one tries to start the app, it fails with the message above.

Please let me know if there is anything that I can provide to support solving this issue. 

Thanks a lot!
Have a great day

**Comment by LZaruba** [comment]
I was able to make it work using the workaround from this macos_ui package issue https://github.com/macosui/macos_ui/issues/424#issuecomment-1544712403

TL;DR
Add the following to your `pubspec.yaml`
```
dependency_overrides:
  macos_ui:
    git:
      url: https://github.com/Mayb3Nots/macos_ui.git
      ref: dev
```

The problem itself is in the macos_ui library not being compatible. Once the version 2.x is out, this project should bump it and all is goood.

**Comment by jiangzongyu** [comment]
any news?

**Comment by m-gug** [comment]
Hi!
@rwbr any chance to get rid of the macos_ui dependency? Due to the developer's personal difficulties, there seems to be little progress on the issue (https://github.com/macosui/macos_ui/issues/423). Tragic and completely understandable, but unfortunately the error is preventing us from upgrading Flutter and thus also from upgrading many other packages.

**Comment by yichaosun** [comment]
UPDATE:
To reproduce, just checkout this repo and try to run the example app with flutter 3.10.x

**Comment by rwbr** [comment]
I'll check that asap

**Comment by rwbr** [comment]
With Flutter 3.22.3 I don't get any problem like this.
